event_operating_unit/views/event_registration_views.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <record model="ir.ui.view" id="view_event_registration_tree_inherit_ou">
+        <field name="name">event.registration.tree</field>
+        <field name="model">event.registration</field>
+        <field name="inherit_id" ref="event.view_event_registration_tree" />
+        <field name="arch" type="xml">
+            <xpath expr="field[@name='state']" position="after">
+                <field name="operating_unit_id" />
```

Comment:
IMO you should add groups="operating_unit.group_multi_operating_unit" on this field.